test/test_tensor_creation_ops.py

```diff
@@ -790,6 +790,75 @@ def compare_helper_(like_fn, t):
                     tp = t.permute(p)
                     compare_helper_(like_fn, tp)
 
+    def _test_special_splits(self, torch_fn, np_fn, dim_range, dim, device, dtype):
+        def get_valid_index_list(sizes, target):
+            ind_list = []
+            curr = 0
+            for i in range(sizes):
+                ind_list += [random.randint(curr, target)]
+                curr = ind_list[-1]
+            return ind_list
+
+        input_sizes = [
+            (0,),
+            (10,),
+            (10, 0),
+            (0, 10),
+            (4, 10),
+            (12, 3),
+            (1, 2, 4),
+            (3, 7, 8),
+            (0, 7, 0),
+            (8, 0, 1),
+            (1, 3, 5, 13),
+            (2, 4, 6, 8),
+            (3, 0, 1, 2),
+            (6, 0, 0, 6),
+            (0, 5, 1, 0)
+        ]
+        for input_size in input_sizes:
+            a_base = self._generate_input(input_size, dtype, device, with_extremal=False)
+            if dim_range[0] <= len(input_size) <= dim_range[1]:
```

Comment:
Rewrite this expression to be clearer -- although maybe it's correct!?

Reply:
Take a closer look at how dim_range is used -- is this the most straightforward way to do this?

Reply:
Yes, actually Python supports this semantics interestingly :) but i re-wrote it more clear version.

For dim_range argument, i would actually want to further discuss further with you. If we check stack operations (hstack, vstack, dstack) implementation, i can see there is an alternative and more systematical implementation for input generation. But i personally liked this implementation as well to give natural filtering to set of inputs.

Hower, I believe generating inputs before entering if statements can cause redundant input generations sometime, which would be slower. So i changed the position we generate inputs.